pkg/vm/builtins/poseidon.go

```diff
@@ -0,0 +1,57 @@
+package builtins
+
+import (
+	"errors"
+	"fmt"
+
+	mem "github.com/NethermindEth/cairo-vm-go/pkg/vm/memory"
+	"github.com/consensys/gnark-crypto/ecc/stark-curve/fp"
+)
+
+const PoseidonName = "poseidon"
+const cellsPerPoseidon = 6
+const inputCellsPerPoseidon = 3
+
+type Poseidon struct{}
+
+func (p *Poseidon) CheckWrite(segment *mem.Segment, offset uint64, value *mem.MemoryValue) error {
```

Comment:
This is due to the builtins specification. Some of them take effect when you write to it, and you call CheckWrite. But since we use shared interface for all the builtins, some of them don't need to use that method